File: LeetCode/Problem1/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        solution = list()
        for num in range(len(nums)):
            for num2 in range(len(nums)):
                if num == num2:
                    pass
                elif nums[num] + nums[num2] == target:
                    solution.append(num)
                    solution.append(num2)
                    return solution

File: LeetCode/Problem1/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> complementMap;
        //unordered map is used here because using a map will use more memory because
        //it holds unnecessary data
        int length = nums.size();

        for(int i = 0; i < length; ++i){
            int complement = target - nums[i];
            if (complementMap.count(complement)) {
                // the .count method returns how many times the item appears in the map
                //This means if the complement exists, the condition evals to true
                return {complementMap[complement], i};
            }
            complementMap[nums[i]] = i;
        }

        return {};
    }
};